src/web/mjs/connectors/Bx117.mjs

```diff
@@ -0,0 +1,78 @@
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+
+export default class Bx117 extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'bx117';
+        super.label = '117漫画网 (bx117)';
+        this.tags = [ 'manga', 'webtoon', 'chinese' ];
+        this.url = 'http://m.bx117.com';
+    }
+    async _getMangaFromURI(uri) {
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'p.txtItme.h1');
+        const id = uri.pathname + uri.search;
+        const title = data[0].textContent.trim();
+        return new Manga(this, id, title);
+    }
+
+    async _getChapters(manga) {
+        const request = new Request(this.url + manga.id, this.requestOptions);
+        const data = await this.fetchDOM(request, 'a.d-nowrap',3);
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.text.trim()
+            };
+        });
+    }
+
+    async _getPages(chapter) {
+        const script = `
+            new Promise(resolve => {
+                resolve(new Array(qTcms_page.total).fill().map((_,ind) => new URL(getPicUrlP(qTcms_S_m_murl,ind+1),qTcms_m_weburl).href));
+            });
+        `;
+        const request = new Request(this.url + chapter.id, this.requestOptions);
+        return await Engine.Request.fetchUI(request, script);
+    }
+
+    async _getMangas() {
+        let mangaList = [];
+        for(let i = 1; i <= 2 ; i++) {
+            for(let page = 1, run = true; run; page++) {
+                let mangas = await this._getMangasFromPage(page,3,i);
+                mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+            }
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page,retries,serial ) {
+        const response = await fetch('http://m.bx117.com/statics/qingtiancms.ashx', {
```

Comment:
Use `this.fetchDOM` instead of re-implementing the functionality.

Reply:
ngl i totally forgot that is what `this.requestOptions` was